plugins/modules/nd_pcv.py

```diff
@@ -154,29 +179,35 @@ def main():
 
     state = nd.params.get("state")
     name = nd.params.get("name")
-    site_name = nd.params.get("site_name")
+    fabric = nd.params.get("fabric")
     insights_group = nd.params.get("insights_group")
     description = nd.params.get("description")
     file = nd.params.get("file")
     manual = nd.params.get("manual")
+    wait_delay = nd.params.get("job_wait_delay")
+    wait_timeout = nd.params.get("job_wait_timeout")
 
     path = "config/insightsGroup"
     if name is None:
         nd.existing = ndi.query_pcvs(insights_group)
-    elif site_name is not None:
-        nd.existing = ndi.query_pcv(insights_group, site_name, name)
+    elif fabric is not None:
+        nd.existing = ndi.query_pcv(insights_group, fabric, name)
 
     if state == "wait_and_query" and nd.existing:
         status = nd.existing.get("analysisStatus")
+        start_time = time.time()
         while status != "COMPLETED":
             try:
-                verified_pcv = ndi.query_pcv(insights_group, site_name, name)
+                verified_pcv = ndi.query_pcv(insights_group, fabric, name)
                 status = verified_pcv.get("analysisStatus")
                 if status == "COMPLETED" or status == "FAILED":
                     nd.existing = verified_pcv
                     break
             except BaseException:
                 nd.existing = {}
+            if wait_timeout and time.time() - start_time >= wait_timeout:
```

Comment:
I'm curious, did you face an intermittent Internal Server Error (500) without introduction of this delay mechanism?

Reply:
I didn't face any specific server errors in my testing. I noticed there was no delay and the module was sending an excessive amount of requests during this while loop. This could be because the ND used for testing didn't have much API traffic and I suspect this module might cause issues in other environments.
I ran into another issue while debugging. The HTTP logs got so large the module failed before the PCV job completed.